units/host_termination.go

```diff
@@ -409,21 +407,9 @@ func (j *hostTerminationJob) Run(ctx context.Context) {
 }
 
 func (j *hostTerminationJob) runHostTeardown(ctx context.Context, settings *evergreen.Settings) (string, error) {
-	if j.host.Distro.Teardown == "" ||
-		j.host.Status == evergreen.HostProvisionFailed ||
-		j.host.SpawnOptions.SpawnedByTask {
-		return "", nil
-	}
-
 	if !j.host.Distro.LegacyBootstrap() {
-		if j.host.Distro.BootstrapSettings.Method == distro.BootstrapMethodUserData {
-			if output, err := j.writeTeardownScript(ctx, settings); err != nil {
-				return output, errors.Wrap(err, "could not put teardown script on host")
-			}
-		}
-
 		// Attempt to run the teardown command through Jasper.
-		output, err := j.tryRunTeardownScript(ctx, settings, func(runScript string) (string, error) {
+		output, err := j.tryRunTeardownScript(func(runScript string) (string, error) {
```

Comment:
I think you should delete `tryRunTeardownScript` as well.

Reply:
I can--I thought it was unrelated since this doesn't have any mention of the distro's teardown script. Do you know how this function works?

Reply:
If teardown.sh is something we also don't want to use in general then we should also get rid of the CLI operation that does teardown.

Reply:
So there's two pieces:

- The app server puts a file called teardown.sh on the host and then runs the CLI command `evergreen host teardown` when the host terminates.
- The CLI command looks for a file called teardown.sh and executes it if it exists.

This means that yes, we should also get rid of the CLI operation too.